### ISSUE TITLE
Fix parsing of path params containing dots

### ISSUE THREAD

**Issue by manosbatsis** [resthub/resthub-spring-stack#188]
In short when using path parameters with dots, the part after the last dot is omitted. To fix this one must add 

``` xml
<bean name="handlerMapping" class="org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerMapping">
    <property name="useSuffixPatternMatch" value="false"></property>
</bean>
```

but this does _not_ work when using

``` xml
 <tx:annotation-driven />
```

which is what resthubContext.xml does. The solution is to replace tx:annotation-driven  with an explicit configuration as described in [1].

[1] http://stackoverflow.com/questions/11658353/spring-requestmapping-not-working-as-expected-if-there-are-extensions-in-url

**Comment by bclozel** [comment]
Did [ticket fixed](https://jira.springsource.org/browse/SPR-8410) the issue in SpringMVC? (see also [this one](https://jira.springsource.org/browse/SPR-7632))

**Comment by manosbatsis** [comment]
Nuh, they actually made this worse as such a path param will now result to an  org.springframework.web.HttpMediaTypeNotAcceptableException: Could not find acceptable representation

**Comment by bmeurant** [comment]
This issue is really outdated, sorry but I will still give you some answers : 

As Brian mentioned, new versions of Spring fixed parts of your problem, but there are still issues : 

**Solution 1** : You can _force_ Spring ignoring the omission of the last dot by defining your mapping with a regexp : 

``` java
@RequestMapping(method = RequestMethod.GET, value = "{param:.+}")
    public @ResponseBody String getTest(@PathVariable String param) {
        return param;
    }
```

as mentioned in : http://stackoverflow.com/questions/16332092/spring-mvc-pathvariable-with-dot-is-getting-truncated

**Solution 2** : If you want some special mapping with dots (in request mapping, not in param) as mentioned in your link http://stackoverflow.com/questions/11658353/spring-requestmapping-not-working-as-expected-if-there-are-extensions-in-url, you should not have anything to do but defining a new method : 

``` java
@RequestMapping(method = RequestMethod.GET, value = "dot.{param}")
    public @ResponseBody String getDotTest(@PathVariable String param) {
        return param;
    }
```

Spring should ignore extension and last dot recognition as shown in #217

**Solution 3** : In fact, the problem comes when you also want to manage extensions and particularily set a cutom list of recognized extensions. In this case, **Solution 1** does not work anymore because regex causes extensions to be integrated to params and ignored by spring recognition scheme.

You will have to provide a custom map of managed extensions and then you will have to override default configuration and this is - that your point - not easily possible with resthub config based on mvc:annotation-driven.

To summarize, I think that we are able to manage most cases with **Solution 1** and **Solution 2** without any overriding. On the other hand, I agree that using mvc:annotation-driven is limiting possibilities in terms of extension and that is not good.

I am currently working on providing a more complete and overridable configuration for resthub but this is not that simple because of custom resthub Exception mappings that are not covered in your original link. 

You can find the complete description in issue #217 and a fix will come soon ...

Again, I know that the issue is very outdated and that you probably already fixed that point but I agree that we need to fix this.

Thanks for raising this point.
